src/olympia/addons/tests/test_update.py

```diff
@@ -416,6 +418,32 @@ def check(self, expected):
                     expected['-'.join([version, mode])]
                 )
 
+    @freeze_time('2018-05-04 11:27:00 UTC')
+    def test_end_to_end(self):
+        # Basic end-to-end test. It's not a django app, so it's a bit awkward
+        # to test this here, but we don't really have a better place to do it.
+        environ = {
+            'QUERY_STRING': ''
+        }
+        self.start_response_call_count = 0
+
+        def start_response_inspector(status, headers):
+            self.start_response_call_count += 1
+            assert status == '200 OK'
+            assert headers == [
+                ('Content-Type', 'application/json'),
+                ('Cache-Control', 'public, max-age=3600'),
+                ('Last-Modified', 'Fri, 04 May 2018 11:27:00 GMT'),
+                ('Expires', 'Fri, 04 May 2018 12:27:00 GMT'),
+                ('Content-Length', '2')
+            ]
+
+        output = update.application(environ, start_response_inspector)
+        assert self.start_response_call_count == 1
+        # Output is an array with a single string containing the body of the
+        # response - in this case an empty json dict.
+        assert output == ['{}']
```

Comment:
An end to end that at least produced a no_updates_output would probably be better - the error could be anything - but I don't suppose it matters, and the other tests cover all the bases already.

Reply:
That's a good point, I was lazy but I'll add basic end-to-end covering the rest.